test/common/src/main/kotlin/org/codice/compliance/verification/ResponseVerifier.kt

```diff
@@ -15,63 +15,41 @@
 package org.codice.compliance.verification
 
 import io.kotlintest.matchers.shouldNotBe
-import org.apache.cxf.rs.security.saml.sso.SSOConstants.*
 import org.codice.compliance.SAMLComplianceException
 import org.codice.compliance.SAMLComplianceExceptionMessage.*
+import org.codice.compliance.saml.plugin.IdpPostResponse
+import org.codice.compliance.saml.plugin.IdpRedirectResponse
+import org.codice.compliance.saml.plugin.IdpResponse
 import org.codice.compliance.utils.TestCommon.Companion.buildDom
-import org.codice.compliance.verification.binding.*
+import org.codice.compliance.verification.binding.BindingVerifier
+import org.codice.compliance.verification.binding.PostVerifier
+import org.codice.compliance.verification.binding.RedirectVerifier
 import org.codice.security.sign.Decoder
 import org.codice.security.sign.Decoder.InflationException
 import org.codice.security.sign.Decoder.InflationException.InflErrorCode
 import org.w3c.dom.Node
 import java.io.IOException
 
-sealed class ResponseVerifier(val response: String, val givenRelayState: Boolean) {
+sealed class ResponseVerifier(val response: IdpResponse) {
 
-    internal fun verifyResponse() : Node {
-        val parsedResponse = parseResponse(response)
+    internal fun verifyResponse(): Node {
+        response.decodedSamlResponse = decodeResponse(response)
+        response.responseDom = buildDom(response)
+        getBindingVerifier(response).verifyBinding()
 
-        val decodedResponse = decodeResponse(parsedResponse)
-        val responseDom = buildDom(decodedResponse)
-        val bindingVerifier = getBindingVerifier(responseDom, parsedResponse, givenRelayState)
-        bindingVerifier.verifyBinding()
-
-        return responseDom
+        return response.responseDom
     }
 
-    abstract protected fun parseResponse(rawResponse: String): Map<String, String>
-    abstract protected fun decodeResponse(parsedResponse: Map<String, String>): String
-    abstract protected fun getBindingVerifier(responseDom: Node, parsedResponse: Map<String, String>, givenRelayState: Boolean): BindingVerifier
+    abstract protected fun decodeResponse(response: IdpResponse): String
+    abstract protected fun getBindingVerifier(response: IdpResponse): BindingVerifier
 
 }
 
```

Comment:
Don't need the `parseResponse` functions because the parsing is done within the new object when receiving the data from the plugin portion.